includes/handlers/class-webmention-handler-meta.php

```diff
@@ -0,0 +1,224 @@
+			}
+			if ( isset( $og['description'] ) ) {
+				$this->webmention_item->set_summary( $og['description'] );
+			}
+			if ( isset( $og['image'] ) ) {
+				$image = $og['image'];
+				if ( is_string( $image ) ) {
+					$this->webmention_item->set_photo( array( $image ) );
+				} elseif ( wp_is_numeric_array( $image ) ) {
+					$this->webmention_item->set_photo( array( $image[0] ) );
+				} else {
+					$this->webmention_item->set_photo( array( $image['secure_url'] ) );
+				}
+			}
+			if ( isset( $og['site_name'] ) ) {
+				$this->webmention_item->set__site_name( $og['site_name'] );
+			}
+
+			if ( isset( $og['longitude'] ) ) {
+				$this->webmention_item->set_location(
+					array(
+						'longitude' => $og['longitude'],
+						'latitude'  => $og['longitude'],
+					)
+				);
+			}
+			if ( isset( $og['type'] ) ) {
+				$type = $og['type'];
+				if ( isset( $meta[ $type ]['tag'] ) ) {
+					$this->webmention_item->set_category( $meta[ $type ]['tag'] );
+				}
+				if ( ! empty( $meta[ $type ]['author'] ) ) {
+					$this->webmention_item->set_author( array( 'name' => $meta[ $type ]['author'] ) );
+				}
+				if ( 'article' === $type ) {
+					if ( isset( $meta['article']['published_time'] ) ) {
+						$this->webmention_item->set_published( new DateTimeImmutable( $meta['article']['published_time'] ) );
+					} elseif ( isset( $meta['article']['published'] ) ) {
+						$this->webmention_item->set_published( new DateTimeImmutable( $meta['article']['published'] ) );
+					}
+					if ( isset( $meta['article']['modified_time'] ) ) {
+						$this->webmention_item->set_updated( new DateTimeImmutable( $meta['article']['modified_time'] ) );
+					} elseif ( isset( $meta['article']['modified'] ) ) {
+						$this->webmention_item->set_updated( new DateTimeImmutable( $meta['article']['modified'] ) );
+					}
+				}
+			}
+		}
+	}
+
+	protected function dublin_core_to_jf2( $meta ) {
+		$item = array();
+		// Then look at Dublin Core Properties
+		if ( isset( $meta['dc'] ) ) {
+			$dc = $meta['dc'];
+			if ( isset( $dc['Title'] ) ) {
+				$item['name'] = $dc['Title'];
+			}
+			if ( isset( $dc['Creator'] ) ) {
+				if ( is_string( $dc['Creator'] ) ) {
+					$this->webmention_item->set_author( array( 'name' => $dc['Creator'] ) );
+				}
+			}
+			if ( isset( $dc['Description'] ) ) {
+				$this->webmention_item->set_summary( $dc['Description'] );
+			}
+			if ( isset( $dc['Date'] ) ) {
+				$this->webmention_item->set_published( new DateTimeImmutable( $dc['Date'] ) );
+			}
+		}
+	}
+
+	protected function add_property( $array, $key, $value ) {
+		if ( ! isset( $array[ $key ] ) ) {
+			$array[ $key ] = $value;
+		} elseif ( is_string( $array[ $key ] ) ) {
+			$array[ $key ] = array( $array[ $key ], $value );
+		} elseif ( is_array( $array[ $key ] ) ) {
+			$array[ $key ][] = $value;
+		}
+		return $array;
+	}
+
+	/**
+	 * Tries to parse OGP Meta tags by hierarchy.
+	 */
+	protected function parse_ogp( $meta ) {
```

Comment:
I think the parse prefix is only confusing, because it does not "parse" the dom document, but filters the meta array. It still feels like a lot of re-arranging of arrays.

Reply:
the function seems to also "parse" DC and any other "namespaced" vocabulary, so I would use a more generic name.